### DISCUSSION TITLE
Is there any possibility to add a custom header to the request?

### DISCUSSION THREAD

**Discussion by salimguerid** [openai/openai-dotnet#315] [Q&A] [answered]
Hello, I need to add a custom header to the request. Is there any option to do so?

**Comment by imDarshanGK** [comment, marked as answer]
Hi @salimguerid,

Yes, you can add a custom header to the request. Here's how you can do it in C#:

var client = new HttpClient();
client.DefaultRequestHeaders.Add("Your-Header-Name", "YourHeaderValue");

If you're using a specific library, check its documentation for the exact method of adding custom headers.

**Reply by salimguerid** [reply]
Thanks!
